pkg/arvo/sys/hoon.hoon

```diff
@@ -1493,10 +1493,12 @@
     =|  {l/(unit) r/(unit)}
     |-  ^-  ?
     ?~  a   &
-    ?&  ?~(l & (gor p.n.a u.l))
-        ?~(r & (gor u.r p.n.a))
-        ?~(l.a & ?&((mor p.n.a p.n.l.a) $(a l.a, l `p.n.a)))
-        ?~(r.a & ?&((mor p.n.a p.n.r.a) $(a r.a, r `p.n.a)))
+    ?&  ?~(l & &((gor p.n.a u.l) !=(p.n.a u.l)))
+        ?~(r & &((gor u.r p.n.a) !=(u.r p.n.a)))
+        ?~  l.a   &
+        &(&((mor p.n.a p.n.l.a) !=(p.n.a p.n.l.a)) $(a l.a, l `p.n.a))
+        ?~  r.a   &
+        &(&((mor p.n.a p.n.r.a) !=(p.n.a p.n.r.a)) $(a r.a, r `p.n.a))
```

Comment:
`&(...)` groups an arbitrary number of expressions, so there's no need to nest those; I'd just  `&((mor ...) !=(...) $(...))`. Sorry for misleading with my earlier comment!

The overall `?&` is pretty substantial now -- I'd be tempted to try different formatting and balance it more (for instance, it might be improved with all the `?~` in tall form). But that's pretty nitpicky, this formatting is also fine as is.

Reply:
Sounds good! thanks @joemfb I've tried with all `?~` in tall form but couldn't come up with something that didn't look that much better compared to the previous one. I'll go ahead with your first suggestion.